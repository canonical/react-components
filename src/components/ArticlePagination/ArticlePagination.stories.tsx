import { Meta, StoryObj } from "@storybook/react";

import ArticlePagination from "./ArticlePagination";

const meta: Meta<typeof ArticlePagination> = {
  component: ArticlePagination,
  tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof ArticlePagination>;

export const Default: Story = {
  name: "Default",

  args: {
    nextLabel: "Consectetur adipisicing elit",
    nextURL: "#next",
    previousLabel: "Lorem ipsum dolor sit amet",
    previousURL: "#previous",
  },
};
