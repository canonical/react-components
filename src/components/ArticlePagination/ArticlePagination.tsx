import classNames from "classnames";
import React from "react";
import type { HTMLProps } from "react";

import type { ClassName, PropsWithSpread } from "types";

export type Props = PropsWithSpread<
  {
    /**
     * Optional classes to add to the wrapping element.
     */
    className?: ClassName;
    /**
     * The URL for the next link.
     */
    nextURL?: string;
    /**
     * The label for the next link.
     */
    nextLabel?: string;
    /**
     * The URL for the previous link.
     */
    previousURL?: string;
    /**
     * The label for the previous link.
     */
    previousLabel?: string;
  },
  HTMLProps<HTMLElement>
>;

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Article pagination](https://docs.vanillaframework.io/patterns/article-pagination/).
 *
 * The article pagination component should be used to navigate from one article to the next, or previous, in chronological order.
 */
const ArticlePagination = ({
  className,
  nextURL,
  nextLabel,
  previousURL,
  previousLabel,
  ...props
}: Props): JSX.Element => {
  return (
    <footer
      className={classNames(className, "p-article-pagination")}
      {...props}
    >
      {previousURL && previousLabel && (
        <a className="p-article-pagination__link--previous" href={previousURL}>
          <span className="p-article-pagination__label">Previous</span>
          <span className="p-article-pagination__title">{previousLabel}</span>
        </a>
      )}
      {nextURL && nextLabel && (
        <a className="p-article-pagination__link--next" href={nextURL}>
          <span className="p-article-pagination__label">Next</span>
          <span className="p-article-pagination__title">{nextLabel}</span>
        </a>
      )}
    </footer>
  );
};

export default ArticlePagination;
