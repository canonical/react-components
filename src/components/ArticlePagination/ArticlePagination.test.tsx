import { render, screen } from "@testing-library/react";
import React from "react";

import ArticlePagination from "./ArticlePagination";

describe("ArticlePagination ", () => {
  it("renders", () => {
    render(
      <ArticlePagination
        nextLabel="Consectetur adipisicing elit"
        nextURL="#next"
        previousLabel="Lorem ipsum dolor sit amet"
        previousURL="#previous"
      />
    );
    expect(screen.getByRole("contentinfo")).toMatchSnapshot();
  });

  it("can add additional classes", () => {
    render(<ArticlePagination className="extra-class" />);
    const pagination = screen.getByRole("contentinfo");
    expect(pagination).toHaveClass("p-article-pagination");
    expect(pagination).toHaveClass("extra-class");
  });
});
