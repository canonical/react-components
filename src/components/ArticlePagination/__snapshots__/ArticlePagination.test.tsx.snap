// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePagination  renders 1`] = `
<footer
  className="p-article-pagination"
>
  <a
    className="p-article-pagination__link--previous"
    href="#previous"
  >
    <span
      className="p-article-pagination__label"
    >
      Previous
    </span>
    <span
      className="p-article-pagination__title"
    >
      Lorem ipsum dolor sit amet
    </span>
  </a>
  <a
    className="p-article-pagination__link--next"
    href="#next"
  >
    <span
      className="p-article-pagination__label"
    >
      Next
    </span>
    <span
      className="p-article-pagination__title"
    >
      Consectetur adipisicing elit
    </span>
  </a>
</footer>
`;
